src/main/java/ch/njol/skript/structures/StructVariables.java

```diff
@@ -66,26 +78,92 @@ public class StructVariables extends Structure {
 		Skript.registerStructure(StructVariables.class, "variables");
 	}
 
-	private final List<NonNullPair<String, Object>> variables = new ArrayList<>();
+	public static class DefaultVariables implements ScriptData {
+
+		private final List<NonNullPair<String, Object>> variables = new ArrayList<>();
+		private final Deque<Map<String, Class<?>[]>> hints = new ArrayDeque<>();
+
+		public DefaultVariables(Collection<NonNullPair<String, Object>> variables) {
+			this.variables.addAll(variables);
```

Comment:
Just create an immutable copy of the parameter then assign it, so there are no more unneeded `Collections#unmodifiableList(List)` calls.

Reply:
`this.variables.addAll` can be replaced with `this.variables = ImmutableList.copyOf(variables);` (or whatever the method is)